core/src/main/java/io/micronaut/core/util/StringUtils.java

```diff
@@ -267,4 +267,23 @@ public static String capitalize(String str) {
         }
         return new String(array);
     }
+
+    /**
+     * Returns the trimmed (left and right) form of the input string. If the string is empty after trimming (or null
+     * was passed in the first place), null is returned, i.e. the input string is reduced to nothing.
+     *
+     * @param string the string to trim
+     * @return the trimmed string or null
+     */
+    public static String trimToNull(String string) {
```

Comment:
return type and parameter should be annotated with `@Nullable`

Reply:
Done.